logical/response.go

```diff
@@ -81,6 +83,36 @@ func (r *Response) Error() error {
 	return nil
 }
 
+func (r *Response) SetError(err error, errorData interface{}) {
```

Comment:
I'm not sure we need this method on the response struct.  I would much rather formatting of the errors being handled by the caller instead of trying to generalize it here.